### DISCUSSION TITLE
Regarding uploading the code developed on my Github repository

### DISCUSSION THREAD

**Discussion by viksof** [community/community#60382] [Programming Help]
### Body

I have some code that I developed to upload on my Github repository in my vitech91 Github account. When I visited the Repository of my Github account on the website link https://github.com/viktech91/viktech91, I am able to see the following screen on my laptop 

![image](https://github.com/community/community/assets/91019704/82a0a153-5f8b-45eb-a902-868931b04804)

May I please know how can I upload the code that I have developed on this Github repository of mine and how can I know more about it?

Can you pls let me know about this?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by GhostOf0days** [comment]
Download [Git](https://git-scm.com/downloads). Also, download [Github Desktop](https://desktop.github.com/). Practice [Git Branching](https://learngitbranching.js.org/?locale=en_US). [Learn Git](https://www.freecodecamp.org/news/introduction-to-git-and-github/). [Learn Github Desktop](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/overview/getting-started-with-github-desktop).

**Comment by ronakjkothari** [comment]
There are many different ways of uploading code to GitHub. The easiest way for a beginner to upload code to GitHub is by using [GitHub Desktop](https://desktop.github.com/) [GitKraken](https://www.gitkraken.com/). I'll be giving instructions for GitHub Desktop as that is easier to use:

1. Download the correct version of [GitHub Desktop](https://desktop.github.com/) depending on your computer software.
2. Login with your GitHub account and follow all instructions on application
3. Clone the repository on the computer.
4. Make desired changes to the repository using your IDE of choice.
5. Navigate back to GitHub desktop where it will list all the changes that have been made.
6. Write a title, description, and press commit.

If you are unsure about downloading software to commit changes, JetBrains IDEs allow you to commit within the IDE. [This article](https://www.jetbrains.com/help/idea/commit-and-push-changes.html) should illustrate how to commit within IntelliJ which can be applied to all other JetBrains IDEs.

Lastly, you can use [Git](https://git-scm.com/downloads) to commit changes using the terminal. The following code should be used (or something similar)

```BASH
git init
git add .
git commit -m "example commit message"
git remote add origin https://github.com/viktech91/viktech91.git
git push -u origin master
```

**Comment by dev-asadali** [comment]
You can upload code using VS code.
To upload code to a remote Git repository from your local machine using Visual Studio Code (VSCode), you can follow these general steps:

1. Initialize Git: If your local project is not already a Git repository, you need to initialize it by running the following command in the terminal within the project's root directory:
```
git init
```

2. Add and commit changes: Stage the files you want to upload to the remote repository using the following command:
```
git add .
```
This command adds all the files in the current directory. If you want to add specific files, replace `.` with the filenames. 

Next, commit your changes with a descriptive message:
```
git commit -m "Your commit message"
```

3. Connect to a remote repository: In VSCode, click on the Git icon in the left sidebar to open the Source Control panel. At the top, you should see an option to publish your repository. Click on "Publish to remote repository" or "Push" to open the remote repository options. 
In Your case remote URL will be [](https://github.com/viktech91/viktech91.git)

4. Configure remote repository: If you haven't added the remote repository yet, click on "Add remote" and enter the repository's URL. Provide a name for the remote, such as "origin."

5. Push changes: After configuring the remote repository, select the branch you want to push from the bottom-left corner of the VSCode window (next to the Git icon). Then, click on the ellipsis (...) in the Source Control panel and choose "Push" to upload your local changes to the remote repository.
![image](https://github.com/community/community/assets/72449631/d800fe33-a305-4455-b6d3-28e39a681bb3)

VSCode will handle the authentication process if necessary, prompting you to provide your Git credentials.

Once the push is complete, your code should be uploaded to the remote Git repository. Make sure you have the necessary permissions and the correct repository URL to push your changes successfully.

Note: These steps assume that you have Git and the necessary Git extensions installed in your VSCode environment.

**Comment by viksof** [comment]
Hi,

Thank you for your email.

Thanks for the information.

May I please know whether it is possible for executing this step from the
operating system (in the form of graphical user interface or gui) instead
of the command line? That is, may I please know whether it is possible for
me to initialize git in my project's root directory from the windows
operating system rather than from the command line?

May i please know how can I initialize git in my project's root directly by
configuring my project's root directly to git from the windows operating
system (through graphical user interface)?

If possible, can you pls explain this step diagramatically (i,e in the form
a picture), so that I can understand it better?

Can you pls let me know about this?

Thanks and best regards,

Vikas

**Comment by dev-asadali** [comment]
Yes you can do this on windows using github desktop . you need to install github desktop on your device.

Here you can see detailed explaination how to init repo in your project and publish it to github.
https://simpledev.io/lesson/initialize-repo-github-desktop-1

After initializing git repo on your project. you have to publish your repo through github desktop as you can see in picture below.
![image](https://github.com/community/community/assets/72449631/383f0405-5747-4cdc-945d-86d200a2a294)

**Comment by ronakjkothari** [comment]
Hey Vikas,

Download [GitHub Desktop](https://desktop.github.com/) or [GitKraken](https://www.gitkraken.com/). For beginners, I would recommend GitHub Desktop.

**Reply by GhostOf0days** [reply]
Honestly, I still use Github Desktop, and I'm not a beginner. However, I only use it for cloning repos and pulling in new changes that I missed. I do everything else in command line.

**Comment by viksof** [comment]
Hi,

Thank you for your email.

Thanks for the information.

I have one more thing to check about the usage of Github desktop (or
whatever it is called) to upload code corresponding to software development
projects.

I have found out a job opening called Software Engineer Sr I in a company
called 'Synopsis' whose website link and job description I am indicated
below for your reference.

https://sjobs.brassring.com/TGnewUI/Search/home/HomeWithPreLoad?partnerid=25235&siteid=5359&PageType=JobDetails&jobid=1993049#jobDetails=1993049_5359

Software Engineer, Sr I

44714BR

INDIA - Bangalore

Job Description and Requirements

*Product and team information:*

Our Software Security and Quality business is all about building faster
secure software. That starts with our static analysis, software composition
analysis, and dynamic analysis so our customers can build security and
quality into the DNA of their code at any stage of the software development
lifecycle and across the supply chain.  We are a team of highly technical
software engineers in a fast-paced dynamic environment building and
maintaining an industry leading Static Analysis tool.

Our *Static Analysis team* is looking for an exceptional developer, with
experience or interest in building and deploying security tools, to join
the Coverity team. The job involves evolving Coverity's state of the art
analysis infrastructure and generally pushing the envelope of static
analysis. In particular, the job will include enhancing the analysis
product for scalable and performant deployment. In this role the developer
will be implementing, testing, and evaluating security checks (or rules)
using the latest Checker API technology and Rust programming language. The
checks specify the types of the defects our static analysis tool reports.
Thus, adding more checks will provide users with better security results
when scanning their code. You will create checks for variety of languages
supported.

The perfect candidate will have a strong passion for improving software
quality and security, enthusiasm about programming languages and compilers,
but foremost,  a strong desire to learn.

*Primary Responsibilities:*
As part of the team, you will be involved in one or more of the following
activities …

   - Design and develop code analysis checkers for various languages.
   - Improve existing and develop new code analysis techniques and
   algorithms.
   - Create test cases for validating different execution scenarios.
   - Perform evaluations of check results using Jenkins automation
   - Contribute to the collection of benchmarks used for check evaluation.

*Qualifications and Experience:*

   - Bachelor's or Master's in computer science or related fields
   - Experience in static analysis and/or related technologies
   - Experience in *RUST or C++* programming (5+ years)
   - Experience with data structures and algorithmic design and analysis
   - Experience with development on Linux/Unix

*Preferred Experience*

   - Demonstrated history of good analytical, debugging, and
   problem-solving skills.
   - Interest or experience in analyzing programming languages, building
   compilers, parsers, or other language analyzers.
   - Interest or experience in dataflow and program analysis.
   - Interest or experience with web application or mobile security,
   familiarity with OWASP Top 10 lists.
   - Ability to work in both independent and collaborative settings.
   - Good written and oral communication skills, for team collaboration and
   product presentations.

Job Category

Engineering

Country

India

Job Subcategory

Software Engineering

Hire Type

Employee

I am thinking to develop code for this job opportunity in Synopsis and
upload the code that I have developed corresponding to this job opportunity
in my Github profile in a public repository. The website link of my Github
profile is as indicated below.

https://github.com/viktech91/viktech91

May I please know how can I upload the code that I develop corresponding to
the above mentioned job opportunity in my Github public repository so that
the recruiters who are reviewing the code that I have developed feel that I
will be suitable for this job and call me for an interview for this job
opportunity in their company and how can I know more about it?

Is there any way to test the code that I have developed in my github
account, so that I can know before hand whether the code that I have
developed will be suitable for presentation to the recruiters or not?

Can you pls let me know about this?

Thanks and best regards,

Vikas

**Comment by viksof** [comment]
Hi,

Thank you for your email.

Thanks for the information.

Irrespective of whatever other people think about me, I am going to go
ahead with full preparations and think about tackling the interview process
for this job role in Synopsis company.

https://sjobs.brassring.com/TGnewUI/Search/home/HomeWithPreLoad?partnerid=25235&siteid=5359&PageType=JobDetails&jobid=1993049#jobDetails=1993049_5359

Software Engineer, Sr I

44714BR

INDIA - Bangalore

Job Description and Requirements

*Product and team information:*

Our Software Security and Quality business is all about building faster
secure software. That starts with our static analysis, software composition
analysis, and dynamic analysis so our customers can build security and
quality into the DNA of their code at any stage of the software development
lifecycle and across the supply chain.  We are a team of highly technical
software engineers in a fast-paced dynamic environment building and
maintaining an industry leading Static Analysis tool.

Our *Static Analysis team* is looking for an exceptional developer, with
experience or interest in building and deploying security tools, to join
the Coverity team. The job involves evolving Coverity's state of the art
analysis infrastructure and generally pushing the envelope of static
analysis. In particular, the job will include enhancing the analysis
product for scalable and performant deployment. In this role the developer
will be implementing, testing, and evaluating security checks (or rules)
using the latest Checker API technology and Rust programming language. The
checks specify the types of the defects our static analysis tool reports.
Thus, adding more checks will provide users with better security results
when scanning their code. You will create checks for variety of languages
supported.

The perfect candidate will have a strong passion for improving software
quality and security, enthusiasm about programming languages and compilers,
but foremost,  a strong desire to learn.

*Primary Responsibilities:*
As part of the team, you will be involved in one or more of the following
activities …

   - Design and develop code analysis checkers for various languages.
   - Improve existing and develop new code analysis techniques and
   algorithms.
   - Create test cases for validating different execution scenarios.
   - Perform evaluations of check results using Jenkins automation
   - Contribute to the collection of benchmarks used for check evaluation.

*Qualifications and Experience:*

   - Bachelor's or Master's in computer science or related fields
   - Experience in static analysis and/or related technologies
   - Experience in *RUST or C++* programming (5+ years)
   - Experience with data structures and algorithmic design and analysis
   - Experience with development on Linux/Unix

*Preferred Experience*

   - Demonstrated history of good analytical, debugging, and
   problem-solving skills.
   - Interest or experience in analyzing programming languages, building
   compilers, parsers, or other language analyzers.
   - Interest or experience in dataflow and program analysis.
   - Interest or experience with web application or mobile security,
   familiarity with OWASP Top 10 lists.
   - Ability to work in both independent and collaborative settings.
   - Good written and oral communication skills, for team collaboration and
   product presentations.

Job Category

Engineering

Country

India

Job Subcategory

Software Engineering

Hire Type

Employee

This is just to let you know about it.

Thanks and best regards,

Vikas
